### PR TITLE
Add description about check constraint

### PR DESCRIPTION
**Pull request by fzzf678**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
 https://github.com/pingcap/tidb/issues/41711
### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: constraints.md [hunk 1]
```diff
@@ -51,20 +51,59 @@ Query OK, 1 row affected (0.03 sec)
...
+`CHECK` 约束是用来限制表中某个字段值必须满足特定条件的一种手段，TiDB 支持 `CHECK` 约束，语法与 MySQL 相同。添加 `CHECK` 约束后，TiDB 会在插入或者更新数据的时候，检查约束条件是否满足，如果不满足，那么会报错。
 
-示例如下：
+`CHECK` 约束语法为：
+```sql
+[CONSTRAINT [symbol]] CHECK (expr) [[NOT] ENFORCED]
+```
+其中`[]`中的内容表示可选项，`CONSTRAINT [symbol]` 表示 `CHECK` 约束的名称。`CHECK (expr)` 表示约束条件, 其中 `expr` 是一个布尔表达式，对于表的每一行，该表达式的计算结果必须为 `TRUE`、`FALSE` 或者 `UNKNOWN` (对于 `NULL` 值)，如果对于某条数据表达式计算结果为 `FALSE`，则表示约束违反。`[NOT] ENFORCED` 表示是否执行约束，可以用来启用或者禁用 `CHECK` 约束。
+
+### 添加约束
```
Comment (CbcWestwolf) [inline constraints.md:62]: 标题后面好像要换行

**Review by CbcWestwolf — APPROVED** [review]

**Review by TonsnakeLin — APPROVED** [review]
LGTM

File: constraints.md (hunk 1 above)
Comment (qiancai) [reply constraints.md:62]: 嗯嗯，是的

File: constraints.md [hunk 2]
```diff
@@ -51,20 +51,62 @@ Query OK, 1 row affected (0.03 sec)
...
+CONSTRAINT `t_chk_1` CHECK ((`a` > 10)) /*!80016 NOT ENFORCED */,
+CONSTRAINT `t_chk_2` CHECK ((1 < `c`))
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
++-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+### 删除约束
+
+删除 `CHECK` 约束时使用名称 `symbol` 来删除指定的约束。语法为:
```
Comment (qiancai) [inline constraints.md:98]:
```suggestion
删除 `CHECK` 约束时，你需要指定需要删除的约束名。

删除约束的语法为:

```

File: constraints.md [hunk 3]
```diff
@@ -51,20 +51,62 @@ Query OK, 1 row affected (0.03 sec)
...
+```sql
+ALTER TABLE t DROP CONSTRAINT t_chk_1;
+```
+
+### 启用/禁用约束
```
Comment (qiancai) [inline constraints.md:103]:
```suggestion
### 启用或禁用 `CHECK` 约束
```

File: constraints.md [hunk 4]
```diff
@@ -51,20 +51,62 @@ Query OK, 1 row affected (0.03 sec)
...
+
+在添加约束的时候，可以指定是否执行约束，如果指定了 `NOT ENFORCED`，那么在插入或者更新数据的时候，不会检查约束条件，如果不指定 `NOT ENFORCED` 或者指定 `ENFORCED`，那么在插入或者更新数据的时候，会检查约束条件。
```
Comment (qiancai) [inline constraints.md:105]:
```suggestion
在为表[添加约束](#添加约束)的时候，可以指定当插入或者更新数据时 TiDB 是否执行约束。

- 如果指定了 `NOT ENFORCED`， 当插入或者更新数据时，TiDB 不会检查约束条件。 
- 如果未指定 `NOT ENFORCED` 或者指定了 `ENFORCED`，当插入或者更新数据的时候，TiDB 会检查约束条件。

```

File: constraints.md [hunk 5]
```diff
@@ -51,20 +51,62 @@ Query OK, 1 row affected (0.03 sec)
...
+`CHECK` 约束是用来限制表中某个字段值必须满足特定条件的一种手段，TiDB 支持 `CHECK` 约束，语法与 MySQL 相同。添加 `CHECK` 约束后，TiDB 会在插入或者更新数据的时候，检查约束条件是否满足，如果不满足，那么会报错。
 
-示例如下：
+`CHECK` 约束语法为：
+```sql
+[CONSTRAINT [symbol]] CHECK (expr) [[NOT] ENFORCED]
+```
+其中`[]`中的内容表示可选项，`CONSTRAINT [symbol]` 表示 `CHECK` 约束的名称。`CHECK (expr)` 表示约束条件, 其中 `expr` 是一个布尔表达式，对于表的每一行，该表达式的计算结果必须为 `TRUE`、`FALSE` 或者 `UNKNOWN` (对于 `NULL` 值)，如果对于某条数据表达式计算结果为 `FALSE`，则表示约束违反。`[NOT] ENFORCED` 表示是否执行约束，可以用来启用或者禁用 `CHECK` 约束。
+
+### 添加约束
```
Comment (qiancai) [inline constraints.md:62]:
```suggestion
### 添加 `CHECK` 约束
```

File: constraints.md [hunk 6]
```diff
@@ -51,20 +51,62 @@ Query OK, 1 row affected (0.03 sec)
...
+  `c` int(11) DEFAULT NULL,
+CONSTRAINT `c1` CHECK ((`b` > `c`)),
+CONSTRAINT `t_chk_1` CHECK ((`a` > 10)) /*!80016 NOT ENFORCED */,
+CONSTRAINT `t_chk_2` CHECK ((1 < `c`))
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
++-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+### 删除约束
```
Comment (qiancai) [inline constraints.md:96]:
```suggestion
### 删除 `CHECK` 约束
```

File: constraints.md [hunk 7]
```diff
@@ -51,20 +51,62 @@ Query OK, 1 row affected (0.03 sec)
...
+删除 `CHECK` 约束时使用名称 `symbol` 来删除指定的约束。语法为:
+```sql
+ALTER TABLE t DROP CONSTRAINT t_chk_1;
+```
+
+### 启用/禁用约束
+
+在添加约束的时候，可以指定是否执行约束，如果指定了 `NOT ENFORCED`，那么在插入或者更新数据的时候，不会检查约束条件，如果不指定 `NOT ENFORCED` 或者指定 `ENFORCED`，那么在插入或者更新数据的时候，会检查约束条件。
+
+除了在添加约束时候指定 `[NOT] ENFORCED`，你还可以在 `ALTER TABLE` 语句中启用或者禁用 `CHECK` 约束，语法为:
```
Comment (qiancai) [inline constraints.md:107]:
```suggestion
除了在添加约束时候指定 `[NOT] ENFORCED`，你还可以在 `ALTER TABLE` 语句中启用或者禁用 `CHECK` 约束，语法为：

```

File: constraints.md [hunk 8]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+
+```sql
+ALTER TABLE t DROP CONSTRAINT t_chk_1;
+```
+
+### 启用或禁用 `CHECK` 约束
+
+在为表[添加约束](#添加约束)的时候，可以指定当插入或者更新数据时 TiDB 是否执行约束。
+
+- 如果指定了 `NOT ENFORCED`， 当插入或者更新数据时，TiDB 不会检查约束条件。
```
Comment (fzzf678) [inline constraints.md:123]:
```suggestion
- 如果指定了 `NOT ENFORCED`，当插入或者更新数据时，TiDB 不会检查约束条件。 
```

File: constraints.md [hunk 9]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+ALTER TABLE t DROP CONSTRAINT t_chk_1;
+```
+
+### 启用或禁用 `CHECK` 约束
+
+在为表[添加约束](#添加约束)的时候，可以指定当插入或者更新数据时 TiDB 是否执行约束。
+
+- 如果指定了 `NOT ENFORCED`， 当插入或者更新数据时，TiDB 不会检查约束条件。 
+- 如果未指定 `NOT ENFORCED` 或者指定了 `ENFORCED`，当插入或者更新数据的时候，TiDB 会检查约束条件。
+
```
Comment (qiancai) [inline constraints.md:125]:
```suggestion
```

File: constraints.md [hunk 10]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
 
 ## `CHECK` 约束
```
Comment (qiancai) [inline constraints.md:53]: https://docs.pingcap.com/zh/tidb/dev/sql-statement-create-table
  和 https://docs.pingcap.com/zh/tidb/dev/sql-statement-alter-table 中的语法图和示例是否需要更新呢

Comment (fzzf678) [reply constraints.md:53]: 不需要，语法没有改变，语法是之前版本就支持的。

File: constraints.md [hunk 11]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+语法说明：
+
+- `[]`中的内容表示可选项。
+- `CONSTRAINT [symbol]` 表示 `CHECK` 约束的名称。
+- `CHECK (expr)` 表示约束条件，其中 `expr` 需要为一个布尔表达式。对于表中的每一行，该表达式的计算结果必须为 `TRUE`、`FALSE` 或 `UNKNOWN` (对于 `NULL` 值) 中的一个。对于某行数据，如果该表达式计算结果为 `FALSE`，则表示违反约束条件。
+- `[NOT] ENFORCED` 表示是否执行约束，可以用于启用或者禁用 `CHECK` 约束。
+
+### 添加 `CHECK` 约束
+
+在 TiDB 中，你可以在 `CREATE TABLE` 或者 `ALTER TABLE` 语句中为表添加 `CHECK` 约束。
```
Comment (Oreoxmt) [inline constraints.md:71]:
```suggestion
在 TiDB 中，你可以在 [`CREATE TABLE`](/sql-statements/sql-statement-create-table.md) 或者 [`ALTER TABLE`](/sql-statements/sql-statement-modify-column.md) 语句中为表添加 `CHECK` 约束。
```

File: constraints.md [hunk 12]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+    CREATE TABLE t(a INT CHECK(a > 10) NOT ENFORCED, b INT, c INT, CONSTRAINT c1 CHECK (b > c));
+    ```
+
+- 在 `ALTER TABLE` 语句中添加 `CHECK` 约束的示例：
+
+    ```sql
+    ALTER TABLE t ADD CONSTRAINT CHECK (1 < c);
+    ```
+
+在添加或者启用 `CHECK` 约束时候，TiDB 会对表中的存量数据进行校验，如果有违反约束的数据存在，那么添加 `CHECK` 约束将会失败并且报错。
```
Comment (Oreoxmt) [inline constraints.md:85]:
```suggestion
在添加或者启用 `CHECK` 约束时，TiDB 会对表中的存量数据进行校验。如果存在违反约束的数据，添加 `CHECK` 约束操作将失败并且报错。
```

File: constraints.md [hunk 13]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+
+在添加 `CHECK` 约束的时，可以指定约束名，也可以不指定约束名，如果不指定约束名，那么 TiDB 会自动生成一个约束名，生成的约束名格式为 `<tableName>_chk_<1, 2, 3...>`。
```
Comment (Oreoxmt) [inline constraints.md:87]:
```suggestion
在添加 `CHECK` 约束时，可以指定约束名，也可以不指定约束名。如果不指定约束名，那么 TiDB 会自动生成一个格式为 `<tableName>_chk_<1, 2, 3...>` 的约束名。
```

File: constraints.md [hunk 14]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+
+### 查看 `CHECK` 约束
+
+你可以通过 `SHOW CREATE TABLE` 查看表中的约束信息。例如：
```
Comment (Oreoxmt) [inline constraints.md:91]:
```suggestion
你可以通过 [`SHOW CREATE TABLE`](/sql-statements/sql-statement-show-create-table.md) 查看表中的约束信息。例如：
```

File: constraints.md [hunk 15]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
++-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+### 删除 `CHECK` 约束
+
+删除 `CHECK` 约束时，你需要指定需要删除的约束名。
+
+例如：
```
Comment (Oreoxmt) [inline constraints.md:114]:
```suggestion
删除 `CHECK` 约束时，你需要指定需要删除的约束名。例如：
```

File: constraints.md [hunk 16]
```diff
@@ -51,20 +51,83 @@ Query OK, 1 row affected (0.03 sec)
...
+
+```sql
+ALTER TABLE t DROP CONSTRAINT t_chk_1;
+```
+
+### 启用或禁用 `CHECK` 约束
+
+在为表[添加 `CHECK` 约束](#添加-check-约束)的时候，可以指定当插入或者更新数据时 TiDB 是否执行约束。
```
Comment (Oreoxmt) [inline constraints.md:122]:
```suggestion
在为表[添加 `CHECK` 约束](#添加-check-约束)的时候，可以指定当插入或者更新数据时 TiDB 是否执行约束检查。
```

File: constraints.md [hunk 17]
```diff
@@ -51,20 +51,81 @@ Query OK, 1 row affected (0.03 sec)
...
- CONSTRAINT min_username_length CHECK (CHARACTER_LENGTH(username) >=4)
-);
-INSERT INTO users (username) VALUES ('a');
-SELECT * FROM users;
+[CONSTRAINT [symbol]] CHECK (expr) [[NOT] ENFORCED]
+```
+
+语法说明：
+
+- `[]`中的内容表示可选项。
```
Comment (Oreoxmt) [inline constraints.md:64]:
```suggestion
- `[]` 中的内容表示可选项。
```

**Review by Oreoxmt — APPROVED** [review]
/lgtm

File: sql-statements/sql-statement-create-table.md [hunk 18]
```diff
@@ -232,7 +232,6 @@ mysql> DESC t1;
 * `COMMENT` 属性不支持 `WITH PARSER` 选项。
 * TiDB 在单个表中默认支持 1017 列，最大可支持 4096 列。InnoDB 中相应的数量限制为 1017 列，MySQL 中的硬限制为 4096 列。详情参阅 [TiDB 使用限制](/tidb-limitations.md)。
 * 当前仅支持 Range、Hash 和 Range Columns（单列）类型的分区表，详情参阅[分区表](/partitioned-table.md)。
-* TiDB 会解析并忽略 `CHECK` 约束，与 MySQL 5.7 相兼容。详情参阅 [`CHECK` 约束](/constraints.md#check-约束)。
```
Comment (fzzf678) [inline sql-statements/sql-statement-create-table.md:235]: updated, PTAL @qiancai

Comment (qiancai) [reply sql-statements/sql-statement-create-table.md:235]: LGTM

File: constraints.md [hunk 19]
```diff
@@ -51,20 +51,81 @@ Query OK, 1 row affected (0.03 sec)
...
+```sql
+ALTER TABLE t DROP CONSTRAINT t_chk_1;
+```
+
+### 启用或禁用 `CHECK` 约束
+
+在为表[添加 `CHECK` 约束](#添加-check-约束)的时候，可以指定当插入或者更新数据时 TiDB 是否执行约束检查。
+
+- 如果指定了 `NOT ENFORCED`，当插入或者更新数据时，TiDB 不会检查约束条件。
+- 如果未指定 `NOT ENFORCED` 或者指定了 `ENFORCED`，当插入或者更新数据的时候，TiDB 会检查约束条件。
```
Comment (Oreoxmt) [inline constraints.md:123]:
```suggestion
- 如果未指定 `NOT ENFORCED` 或者指定了 `ENFORCED`，当插入或者更新数据时，TiDB 会检查约束条件。
```

**Review by Oreoxmt — APPROVED** [review]
/lgtm

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**